src/parallel/KDNode.h

```diff
@@ -107,27 +107,32 @@ class KDNode {
 	 */
 	unsigned int getNumMaxProcs();
 
-	double calculateAvgLoadPerProc() {
+	double calculateAvgLoadPerProc() const {
 		return _load / ((double) _numProcs);
 	}
 
-	void calculateExpectedDeviation(std::vector<double>* accumulatedProcessorSpeeds = nullptr) {
+	/**
+	 * Calculates a lower bound for the expected deviation.
+	 * @param accumulatedProcessorSpeeds
+	 */
+	void calculateDeviationLowerBound(std::vector<double>* accumulatedProcessorSpeeds = nullptr) {
 		double meanProcessorSpeed[] = { 1., 1. };
-		double averagedMeanProcessorSpeed = 1.;
-		if (accumulatedProcessorSpeeds != nullptr && accumulatedProcessorSpeeds->size() != 0) {
+        double totalMeanProcessorSpeed = 1.;
+		if (accumulatedProcessorSpeeds != nullptr and not accumulatedProcessorSpeeds->empty()) {
```

Comment:
```suggestion
		if (accumulatedProcessorSpeeds and not accumulatedProcessorSpeeds->empty()) {
```

Reply:
Also: you here replaced an `&&` with `and`. While I like this it should probably be consistent throughout the codebase to reduce confusion (I used "consistent" in the context of ls1 lol).

Reply:
:P